cnd/src/btsieve/bitcoin/bitcoind_connector.rs

```diff
@@ -35,23 +37,35 @@ impl BitcoindConnector {
             .join(&format!("{}.hex", block_hash))
             .expect("building url should work")
     }
+
+    async fn chain_info(&self) -> anyhow::Result<ChainInfo> {
+        let url = &self.chaininfo_url;
+        let chain_info = self
+            .client
+            .get(url.clone())
+            .send()
+            .await
+            .context(GetRequestFailed(url.clone()))?
```

Comment:
We might want to use `with_context` here to avoid the allocation in case we don't actually fail.

Reply:
Yes!